hyperdome_client/threads.py

```
@@ -72,20 +72,232 @@ def run(self):
+                 session: requests.Session):
+        super(GetUidTask, self).__init__()
+        self.server = server
+        self.session = session
 
     def run(self):
-        self.mode.common.log('WebThread', 'run')
-        self.mode.app.choose_port()
-        self.mode.web.start(self.mode.app.port,
-                            self.mode.app.stay_open)
+        try:
+            uid = get_uid(self.server, self.session)
+            self.signals.success.emit(uid)
+        except requests.RequestException:
+            self.signals.error.emit("couldn't get UID")
+
+
+class StartChatTask(QtCore.QRunnable):
+    """
+    Signin counselor, or request counselor session if user
+    """
+    signals = TaskSignals()
+
+    def __init__(self,
+                 server: Server,
+                 session: requests.Session,
+                 uid: str):
+        super(StartChatTask, self).__init__()
+        self.server = server
+        self.session = session
+        self.uid = uid
+
+    def run(self):
+        try:
+            self.signals.success.emit(start_chat(
+                self.server, self.session, self.uid))
+        except requests.RequestException:
+            self.signals.error.emit("Couldn't start a chat session")
+
+
+class GetMessagesTask(QtCore.QRunnable):
+    """
+    retrieve new messages on a fixed interval
+    """
+    signals = TaskSignals()
+
+    def __init__(self, session: requests.Session, server: Server, uid: str):
+        super(GetMessagesTask, self).__init__()
+        self.session = session
+        self.server = server
+        self.uid = uid
+
+    def run(self):
+        try:
+            new_messages = get_messages(self.server, self.session, self.uid)
+            self.signals.success.emit(new_messages)
+        except requests.HTTPError:
+            self.signals.error.emit("Counselor not in chat")
+        except requests.RequestException:
+            self.signals.error.emit("Error in get messages request")
```

Same thing with swallowing exceptions, and this one should DEFINITELY be logged

---

logs are boring.